src/gui/autoexpandabledialog.cpp

```diff
@@ -57,6 +57,14 @@ QString AutoExpandableDialog::getText(QWidget *parent, const QString &title, con
     d.m_ui->textEdit->setEchoMode(mode);
     d.m_ui->textEdit->setInputMethodHints(inputMethodHints);
 
+    d.m_ui->textEdit->selectAll();
+    if (excludeExtension) {
+        const int lastDotIndex = text.lastIndexOf('.');
```

Comment:
Should we account for tar files? eg foo.tar.gz?
Something like this:
```c++
int lastDotIndex = text.lastIndexOf('.');

if ((lastDotIndex > 3) && (text.mid(lastDotIndex - 4, 4).toLower() == ".tar"))
    lastDotIndex -= 4;
```

Reply:
Is `(lastDotIndex > 3)` needed? I tested without it and it seems to be the same.
Also i don't suppose we care that if a file is named `.tar.<something>` it selects everything right?

Reply:
It is for checking if the index is bigger than `-1` and to ensure that there are at least 4 chars (length of ".tar") + 1 in the remaining string.
I don't know what happens if you pass negative pos value into mid().

Reply:
>I don't know what happens if you pass negative pos value into mid().

Nothing bad, i tested it thoroughly and it's exactly the same with and without `lastDotIndex > 3`

Reply:
Done.